File: java/209_MinimumSizeSubarraySum.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        int len = nums.length;
        if (len == 0) {
            return 0;
        }
        int res = Integer.MAX_VALUE;
        int i = 0, j = 0;
        int sum = 0;
        while (j < len) {
            sum += nums[j];
            while (sum >= s) {
                res = Math.min(res, j - i + 1);
                sum -= nums[i];
                i++;
            }
            j++;
        }
        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: c++/209_MinimumSizeSubarraySum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int res = INT_MAX;
        for (int i = 0, j = 0, sum = 0; j < nums.size(); j++) {
            sum += nums[j];
            while (sum - nums[i] >= target) sum -= nums[i++];
            if (sum >= target) {
                res = min(res, j - i + 1);
            }
        }
        if (res == INT_MAX) res = 0;
        return res;
    }
};